src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
+        return d->exchanges[name].data();
+
     QAmqpExchange *exchange = new QAmqpExchange(channelNumber, this);
     d->methodHandlersByChannel[exchange->channelNumber()].append(exchange->d_func());
     connect(this, SIGNAL(connected()), exchange, SLOT(_q_open()));
     connect(this, SIGNAL(disconnected()), exchange, SLOT(_q_disconnected()));
     exchange->d_func()->open();
 
-    if (!name.isEmpty())
+    if (!name.isEmpty()) {
         exchange->setName(name);
+        d->exchanges[name] = exchange;
```

Comment:
only adding it to the hash if it has a name? I think the ```d->exchanges[name] = exchange;``` should go before the ```return exchange```, no difference, you always have a QString to key against